client/src/main/java/io/pravega/client/stream/impl/ReaderGroupStateManager.java

```diff
@@ -339,15 +348,15 @@ private boolean shouldAcquireSegment() throws ReaderNotInReaderGroupException {
             if (acquireTimer.hasRemaining()) {
                 return false;
             }
-            if (state.getCheckpointForReader(readerId) != null) {
-                return false;
-            }
             if (state.getNumberOfUnassignedSegments() == 0) {
                 if (doesReaderOwnTooManySegments(state)) {
                     acquireTimer.reset(calculateAcquireTime(readerId, state));
                 }
                 return false;
             }
+            if (state.getCheckpointForReader(readerId) != null) {
```

Comment:
I optimized this method, I don't think this reordering is needed anymore.

Reply:
It does if `acquireTimer.hasRemaining()` is `false` quite often, which seems to be the case. By doing this, we are preventing the execution of `getCheckpointForReader()` thanks to the modification in the implementation of `doesReaderOwnTooManySegments`. If it finally turns out that there is a problem in the way `acquireTimer` is being set, then we can fix it and revert this change.